targets/TARGET_STM/TARGET_STM32H7/clock_cfg/TARGET_STM32H7_280MHZ/system_clock.c

```diff
@@ -96,20 +124,50 @@ MBED_WEAK uint8_t SetSysClock_PLL_HSE(uint8_t bypass)
     } else {
         RCC_OscInitStruct.HSEState = RCC_HSE_ON;
     }
-#if HSE_VALUE==8000000
-    RCC_OscInitStruct.PLL.PLLM = 4;   // 2 MHz
-    RCC_OscInitStruct.PLL.PLLN = 280; // 560 MHz
+
+
+    if(HSE_VALUE % 2000000 == 0)
+    {
+        // Clock divisible by 2.  Divide down to 2MHz and then multiply up again.
+        RCC_OscInitStruct.PLL.PLLM = (HSE_VALUE / 2000000); // PLL1 input clock = 2MHz
+        RCC_OscInitStruct.PLL.PLLRGE = RCC_PLL1VCIRANGE_1; // PLL1 input clock is between 2 and 4 MHz
+
+#if MBED_CONF_TARGET_ENABLE_OVERDRIVE_MODE
+        RCC_OscInitStruct.PLL.PLLN = 280; // PLL1 internal (VCO) clock = 560 MHz
+#else
+        RCC_OscInitStruct.PLL.PLLN = 225; // PLL1 internal (VCO) clock = 450 MHz
+#endif
+    }
+    else if(HSE_VALUE % 5000000 == 0)
+    {
+        RCC_OscInitStruct.PLL.PLLM = (HSE_VALUE / 5000000); // PLL1 input clock = 5MHz
+        RCC_OscInitStruct.PLL.PLLRGE = RCC_PLL1VCIRANGE_2; // PLL1 input clock is between 4 and 8 MHz
+
+#if MBED_CONF_TARGET_ENABLE_OVERDRIVE_MODE
+        RCC_OscInitStruct.PLL.PLLN = 112; // PLL1 internal (VCO) clock = 560 MHz
 #else
-#error Unsupported externall clock value, check HSE_VALUE define
+        RCC_OscInitStruct.PLL.PLLN = 90; // PLL1 internal (VCO) clock = 450 MHz
 #endif
+    }
```

Comment:
can you fix the style to be consistent in this file:
`{` attached on the same line).